src/s3fetchers.ts

```diff
@@ -0,0 +1,113 @@
+import { Readable } from 'stream';
+
+import { S3Client, ListObjectsV2Command, GetObjectCommand } from "@aws-sdk/client-s3";
+
+import { BlockHeight, LakeConfig, Block, Shard, StreamerMessage } from './types';
+import { prependZeroes, parseBody } from './utils';
+
+// Queries the list of the objects in the bucket, grouped by "/" delimiter.
+// Returns the list of blocks that can be fetched
+export async function listBlocks(
+  client: S3Client,
+  bucketName: string,
+  startAfter: BlockHeight,
+): Promise<BlockHeight[]> {
+  try {
+    const data = await client.send(
+        new ListObjectsV2Command({
+          Bucket: bucketName,
+          MaxKeys: 10,
+          Delimiter: "/",
+          StartAfter: prependZeroes(startAfter),
+          RequestPayer: "requester",
+        })
+    );
+    const blockHeights = data.CommonPrefixes.map(p => parseInt(p.Prefix.split('/')[0]));
+    return blockHeights;
+  } catch (err) {
+    console.log("Error", err);
```

Comment:
I think it will also be a good practice to specify the point of failure - in this case `listBlocks` function invokation.